helloworld/notebooks/OpTitanicSimple.ipynb

```diff
@@ -0,0 +1,336 @@
+    "import com.salesforce.op.stages.impl.classification.BinaryClassificationModelsToTry._\n",
+    "\n",
+    "val prediction =\n",
+    "      BinaryClassificationModelSelector.withTrainValidationSplit(\n",
+    "        modelTypesToUse = Seq(OpLogisticRegression)\n",
+    "      ).setInput(survived, finalFeatures).getOutput()\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "val evaluator = Evaluators.BinaryClassification().setLabelCol(survived).setPredictionCol(prediction)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Notice that everything we’ve done so far has been purely at the level of definitions. We have defined how we would like to extract our raw features from data of type `Passenger`, and we have defined how we would like to manipulate them. In order to actually manifest the data described by these features, we need to add them to a workflow and attach a data source to the workflow:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import spark.implicits._ // Needed for Encoders for the Passenger case class\n",
+    "import com.salesforce.op.readers.DataReaders\n",
+    "\n",
+    "val trainFilePath = \"/home/beakerx/helloworld/src/main/resources/TitanicDataset/TitanicPassengersTrainData.csv\"\n",
+    "    // Define a way to read data into our Passenger class from our CSV file\n",
+    "val trainDataReader = DataReaders.Simple.csvCase[Passenger](\n",
+    "      path = Option(trainFilePath),\n",
+    "      key = _.id.toString\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "val workflow =\n",
+    "      new OpWorkflow()\n",
+    "        .setResultFeatures(survived, prediction)\n",
+    "        .setReader(trainDataReader)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "When we now call ‘train’ on this workflow, it automatically computes and executes the entire DAG of Stages needed to compute the features survived, prediction, rawPrediction, and prob, fitting all the estimators on the training data in the process. Calling score on the fitted workflow then transforms the underlying training data to produce a DataFrame with the all the features manifested. The score method can optionally be passed an evaluator that produces metrics."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "val fittedWorkflow = workflow.train()\n",
+    "println(s\"Summary: ${fittedWorkflow.summary()}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After model has been fitted we use `scoreAndEvaluate()` function to evaluate the metrics"
```

Comment:
it would be nice to mention that you can change out the data before doing this by either setting a new path or a new reader

Reply:
changed